fingerprint2.js

```diff
@@ -52,7 +52,18 @@
       return target;
     },
     get: function(done){
-      var keys = [];
+      var that = this;
+      var keys = {
+          data: [],
+          push: function(pair){
```

Comment:
I'd rather see this as an extra method/function with a meaningful name, something like `addPreprocessedComponent` or `preprocessAndAdd`. Also `tmpX` aren't good variable names at all, please use something more meaningful here.

I also don't like that you created a "looks-like-an-array-but-isn't" object here -- either it looks like an array, then it should be an array too, or make it look completely different. Otherwise people will get confused.